packages/salesforcedx-vscode-apex/src/views/testRunner.ts

```diff
@@ -77,7 +79,20 @@ export class ApexTestRunner {
   }
 
   public getTempFolder(): string {
-    return os.tmpdir();
+    if (vscode.workspace && vscode.workspace.workspaceFolders) {
+      const sfdxDir = path.join(
+        vscode.workspace.workspaceFolders[0].uri.fsPath,
+        '.sfdx'
+      );
+      const apexTestPath = path.join(sfdxDir, 'apexTests');
+
+      if (!pathExists.sync(apexTestPath)) {
+        mkdir('-p', apexTestPath);
+      }
+      return apexTestPath;
+    } else {
+      throw new Error();
```

Comment:
Could you add an error message to this error?

Reply:
Yeah, I'll add in a message for the error.